### PR TITLE
Add Github's suggested community files

### LINKED ISSUE #639: Update GitHub Community profile
We should complete the checklist on https://github.com/dafny-lang/dafny/community

This includes:

- Adding a code of conduct
- Adding a "contributing" document
- Creating a GitHub issues template
- Creating a pull request template

### PR DESCRIPTION
**Pull request by cpitclaudel**
Closes #639.  I didn't add multiple issue templates because they solve a problem that we don't have: the bug reports we get are pretty good, so there's no need to burden contributors with additional templates and click-through menus.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+- [Stack Overflow](https://stackoverflow.com/questions/tagged/dafny), an online Q&A website, for help with writing and proving Dafny programs.
+- [`dafny-lang/ide-vscode`](https://github.com/dafny-lang/ide-vscode), Dafny's VSCode plugin, for issues or feature requests specific to the plugin itself (however, issues with Dafny's LSP server should be reported here).
+- [`boogie-org/boogie-friends`](https://github.com/boogie-org/boogie-friends/), Dafny's Emacs mode, for Emacs-specific problems.
+- [`boogie-org/boogie`](https://github.com/boogie-org/boogie), Dafny's intermediate verification language, for questions and issues specific to Boogie.
+
+All other pull requests and issues can be submitted here.
+
+- For issues, please include your Dafny version number, any relevant code, and expected results.
+
+- For pull requests, consider updating <../RELEASE_NOTES.md> if your change is user-visible, and add tests if possible.
```
Comment (MikaelMayer) [inline .github/CONTRIBUTING.md:16]: Will this create a link? I don't see any in the preview

**Review by MikaelMayer — APPROVED** [review]

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (fabiomadge) [inline .github/CONTRIBUTING.md:16]:
```suggestion
- For pull requests, consider updating [`RELEASE_NOTES.md`](RELEASE_NOTES.md) if your change is user-visible, and add tests if possible.
```
This _might_ work.

Comment (cpitclaudel) [reply .github/CONTRIBUTING.md:16]: Isn't that the wrong path?

Comment (fabiomadge) [reply .github/CONTRIBUTING.md]: I was hoping that is would be relative to the root, but I just tried it and you're right.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+- For issues, please include your Dafny version number, any relevant code, and expected results.
+
+- For pull requests, consider updating [`RELEASE_NOTES.md`](../RELEASE_NOTES.md) if your change is user-visible, and add tests if possible.
+
+  - Dafny's integration tests are in [`Test`](../Test).  PRs that fix issues reported on GitHub should include a test in [`Test/git-issues/`](../Test/git-issues/).
+
+    Each `.dfy` file in `Test/` a test, with a  [`lit`](https://llvm.org/docs/CommandGuide/lit.html) header describing how to run it and a `.expect` file indicating the expected output.  See [`Test/README.md`](../Test/README.md) for more info on running Dafny' integration tests.
+
+  - Dafny's unit tests are in various `.Test` directories in [`Source`](../Source).
```
Comment (fabiomadge) [inline .github/CONTRIBUTING.md:22]:
```suggestion
  - Dafny's unit tests are in various `*.Test` directories in [`Source`](../Source).
```

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  Our CI is configured to run all tests when you create a PR.  To run tests locally, use `dotnet test Source/Dafny.sln` (but note that running the tests for our compiler backends requires installing lot of additional software).
```
Comment (fabiomadge) [inline .github/CONTRIBUTING.md:24]:
```suggestion
  Our CI is configured to run all tests when you create a PR.  To run tests locally, use `dotnet test Source/Dafny.sln` (but note that running the tests for our compiler backends requires installing lots of additional software).
```

**Review by keyboardDrummer — DISMISSED** [review]

**Comment by keyboardDrummer** [comment]
Nice 💯

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (cpitclaudel) [reply .github/CONTRIBUTING.md]: Fixed, sorry!  I missed this comment the first time around.

**Review by fabiomadge — APPROVED** [review]

**Pull request merged**